x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -245,17 +242,21 @@ static void getMemoryAndProcessors(
             final int numMissingProcessors = numMissingAllocations * numberOfThreadsPerAllocation;
             int numExistingProcessorsToBeUsed = Math.min(numMissingProcessors, numberOfAvailableProcessors);
 
-            if (AssignmentState.STARTING.equals(assignment.getAssignmentState()) && assignment.getNodeRoutingTable().isEmpty()) {
-                // this state occurs when the assignment has just been added and has not been assigned to any nodes
-                logger.debug(
-                    () -> format(
-                        "trained model [%s] lacks assignment , memory required [%d]",
-                        modelAssignment.getKey(),
-                        estimatedMemoryUsage
-                    )
-                );
-                extraPerNodeModelMemoryBytes = Math.max(extraPerNodeModelMemoryBytes, estimatedMemoryUsage);
-                extraModelMemoryInBytes += estimatedMemoryUsage;
+            if (assignment.getNodeRoutingTable().isEmpty() == false
+                && assignment.getNodeRoutingTable().values().stream().allMatch(r -> r.getState().consumesMemory() == false)) {
+                // Ignore states that don't consume memory, for example all allocations are failed or stopped
+                // if the node routing table is empty, then it will match the above condition, but it needs to be handled in the next branch
+                continue;
+            } else {
+
+                if (assignment.getNodeRoutingTable().isEmpty() == false) {
+                    // if the routing table is non-empty, this is an existing model
+                    existingModelMemoryBytes += estimatedMemoryUsage;
+                } else {
+                    // only increase memory requirements for new models
+                    extraPerNodeModelMemoryBytes += estimatedMemoryUsage;
+                    extraModelMemoryInBytes += estimatedMemoryUsage;
+                }
```

Comment:
I don't think that this is the correct way to handle memory usage, but as discussed in other comments, the architecture of this system doesn't allow for accurate memory calculations

Reply:
I changed this to use the old calculation, so at least it isn't unnecessarily different, even if its not correct.